.github/workflows/pr-slack-notification.yaml

```diff
@@ -0,0 +1,287 @@
+        run: |
+          mkdir -p cache
+          if [ -f cache/notified_prs.json ]; then
+            echo "Cache restored successfully."
+            cat cache/notified_prs.json
+          else
+            echo "Cache not restored or file does not exist.."
+          fi
+
+      - name: Fetch all opened pull request details using condition
+        id: fetch_all_pull_requests_using_condition
+        run: |
+          # The number of results per page is limited to a maximum of 100. We need "pagination", if we require more than this limit, but for open PRs, it will be within the limit 100.
+          pr_infos=$(curl -s -H "Authorization: token ${{ env.GH_TOKEN }}" \
+          "${{ env.API_URL }}?state=open&direction=desc&per_page=100")
+          
+          echo "List of Currently Opened PRs: "
+          echo "$pr_infos" | jq '.[] | {number, updated_at, draft}'
+
+          # Load previous PR data if exists
+          if [ -f cache/notified_prs.json ]; then
+            previous_prs=$(cat cache/notified_prs.json)
+          else
+            previous_prs="[]"
+          fi
+
+          pr_list=""
+          new_notified_prs="[]"
+          notify=false
+
+          for pr_info in $(echo "$pr_infos" | jq -r '.[] | @base64'); do
+            _jq() {
+              echo "$pr_info" | base64 --decode | jq -r "${1}"
+            }
+
+            pr_number=$(_jq '.number')
+            pr_title=$(_jq '.title')
+            pr_user=$(_jq '.user.login')
+            pr_url=$(_jq '.html_url')
+            pr_draft=$(_jq '.draft')
+            pr_created_at=$(_jq '.created_at')
+            pr_updated_at=$(_jq '.updated_at')
+
+            pr_data=$(jq -n --arg number "$pr_number" --arg updated_at "$pr_updated_at" '{number: $number, updated_at: $updated_at}')
+            new_notified_prs=$(echo "$new_notified_prs" | jq --argjson pr_data "$pr_data" '. += [$pr_data]')
+
+            # Check if the PR is new or updated
+            previous_pr=$(echo "$previous_prs" | jq --arg number "$pr_number" '.[] | select(.number == $number)')
+            if [ -z "$previous_pr" ] || [ "$(echo "$previous_pr" | jq -r '.updated_at')" != "$pr_updated_at" ]; then
+              draft_status=""
+              if [ "$pr_draft" = "true" ]; then
+                draft_status="\n*Draft*: true"
+              fi
+
+              pr_list="${pr_list}\n*Pull Request* #${pr_number}: ${pr_title}\n*Created by*: ${pr_user}\n*URL*: ${pr_url}${draft_status}\n*Created At*: ${pr_created_at}\n*Last Updated At*: ${pr_updated_at}\n"
+              notify=true
+            fi
+          done
+          
+          echo "List of PRs that need to be sent to Slack now:"
+          echo "$pr_list"
+
+          # Save current PR data for future comparison
+          echo "$new_notified_prs" > cache/notified_prs.json
+
+          if [ "$notify" = true ]; then
+            echo -e "$pr_list" > pr_list.txt
+            echo "::set-output name=notify::true"
```

Comment:
There is a warning in the workflow file - `The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files.`. So it is better to change set-output to the GITHUB_OUTPUT environment files. Ref - https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Reply:
@anusreelakshmi934 , I modified the code to use `$GITHUB_ENV` , because this is using for pass values from one step to another step within the same job. Now I am not getting the warning.

Example of recent run: https://github.com/vaisakhkannan/liberty-tools-intellij/actions/runs/9793149679/job/27040487378